plugins/aap-backend/src/providers/AapResourceEntityProvider.ts

```diff
@@ -31,75 +35,79 @@ export class AapResourceEntityProvider implements EntityProvider {
   private connection?: EntityProviderConnection;
 
   static fromConfig(
-    configRoot: Config,
-    options: {
+    deps: {
+      config: Config;
       logger: LoggerService;
-      schedule?: TaskRunner;
-      scheduler?: PluginTaskScheduler;
     },
+
+    options:
+      | { schedule: SchedulerServiceTaskRunner }
+      | { scheduler: SchedulerService },
   ): AapResourceEntityProvider[] {
-    const providerConfigs = readAapApiEntityConfigs(configRoot);
+    const { config, logger } = deps;
+
+    const providerConfigs = readAapApiEntityConfigs(config);
 
     return providerConfigs.map(providerConfig => {
       let taskRunner;
-      if (options.scheduler && providerConfig.schedule) {
+      if ('scheduler' in options && providerConfig.schedule) {
         taskRunner = options.scheduler.createScheduledTaskRunner(
           providerConfig.schedule,
         );
-      } else if (options.schedule) {
+      } else if ('schedule' in options) {
         taskRunner = options.schedule;
       } else {
-        throw new Error(
+        throw new InputError(
           `No schedule provided neither via code nor config for AapResourceEntityProvider:${providerConfig.id}.`,
```

Comment:
Since in the new backend system schedule is provided only through config and not code, would it be better to update the Error message to something like this?
```suggestion
          `No schedule provided via config for AapResourceEntityProvider:${providerConfig.id}.`,
```

Reply:
Good point. We could likely remove the option to add it through code entirely, since a default schedule is applied if none is provided.